pages/resources/[page].js

```diff
@@ -48,28 +52,36 @@ function Resources() {
   const [totalPages, setTotalPages] = useState(currentPage);
   const [allCategories, setAllCategories] = useState([]);
   const [allLanguages, setAllLanguages] = useState([]);
-  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [afterLoginRedirectPath, setLoginRedirectPath] = useState(router.asPath);
 
-  const handleLogin = value => loginUser(value);
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const openLoginModal = (redirectPath = router.asPath) => {
+    setLoginRedirectPath(redirectPath);
+    setIsModalOpen(true);
+  };
+  const closeLoginModal = () => setIsModalOpen(false);
 
-  const handleLoginSuccess = ({ token }) => {
+  const onLogin = value => loginUser(value);
+  const onLoginSuccess = ({ token }) => {
     setAuthCookies({ token });
+    router.push(afterLoginRedirectPath);
   };
 
-  const handleVote = (voteDirection, id, setUpVotes, setDownVotes) => {
+  const onVote = (voteDirection, id, setUpVotes, setDownVotes) => {
     setErrorMessage(null);
-    if (!hasValidAuthToken()) {
-      setIsModalOpen(true);
-      return;
+
+    if (!isLoggedIn) {
+      openLoginModal();
```

Comment:
When I click a `thumbs up` or `thumbs down` on a resource, it doesn't prompt me to log in.. which I thought it would do based on this code. Is that working for you?

Reply:
... hmmm it was for me. Maybe I had a regression I didn't notice.

Reply:
Does it work for u on prod?

Reply:
It works for me on prod... I'm not sure, maybe it's specific to vercel or something?

Reply:
No, I've messed something up on this branch I mean since integration tests are failing. Just didnt realize during local dev

Reply:
No worries, happy to QA it and find those bugs 😄